README.md

```diff
@@ -35,66 +39,85 @@ project's tests. (Note that `build` depends on `check` which depends on the `jac
-- average coverage in a package
-- coverage in a class
-- coverage in a file
+Code coverage requirements are specified separately for the different "realms" reported by Jacoco:
+- average coverage in a report, specified by the `report` keyword
+- average coverage in a package, specified by the `_package` keyword
+- coverage in a class, specified by the `_class` keyword
+- coverage in a file, specified by the `file` keyword
 
-Scopes are specified either by their exact name as it appears in the Jacoco XML report, for example package scope
-`"org/product/module"`, file scope `"MyClass.java"`, class scope `"org/product/module/MyCLass"`), or by a regular
-expression matching those scope names, for example `~"org.*"` for all packages or class scopes starting with `org`.
+Note that the keywords for package and class realms require a `_` prefix since they are reserved words in Groovy. Each 
+realm contains a number of so-called "scopes": 
+- The file realm contains one scope for every source file, named by file name, e.g., "MyClass.java"
+- The class realm contains one scope for every (top-level, inner, or anonymous) class, named by fully qualified class
+name, e.g., "org/package/MyClass$1"
+- The package realm contains one scope for every package, named by the package name, e.g., "org/package"
+- The report realm contains exactly one scope which is usually derived from the name of the corresponding Gradle (sub-)
+project
 
-Coverage requirements are with respect to one of the Jacoco-defined types of coverage, e.g., lines covered or branches
-covered.
+Scopes can be specified by exact name, e.g., `"MyClass.java"`, or by a Groovy regex pattern matching an arbitrary number
+of scopes, e.g., `~'org/package.*'` for all sub-packages of "org/mypackage".
 
-The following examples describe the syntax:
+Coverage requirements are written in terms of a realm, a scope specification, and one (or several) of the Jacoco-defined
+types of coverage, e.g., lines covered or branches covered. The following examples describe the syntax:
 
-    jacocoCoverage {
-        // Minimum code coverage of 50% for all scopes (files, classes, packages, reports) and coverage types.
-        threshold 0.5
 
-        // Minimum 'branch' coverage of 30% for all scopes.
-        // Available coverage types: BRANCH, CLASS, COMPLEXITY, INSTRUCTION, LINE, METHOD.
-        threshold 0.3, BRANCH
+        // 90% coverage required for a WellTestedClass
+        _class 0.9, "org/package/WellTestedClass"
 
-        // Minimum average coverage of 30% in given package.
-        threshold 0.3, "org/company/module"
+        // 10% coverage for all sub-packages of org/package
+        _package 0.1, ~"org/package.*"
+        report 0.8 // 80% coverage required at the report level, i.e., for at average across the project
 
-        // Minimum average coverage of 50% in report "my-project; the report name is usually the Gradle project name.
-        threshold 0.5, "my-project" 
+
+    jacocoCoverage {
```

Comment:
i like this example block, but i leave it wondering what the actual spec is for each one.

- `file` uses a `file THRESHOLD, TYPE [, FILE_FILTER]
- `_package` uses `_package THRESHOLD, PACKAGE_NAME`
- etc.

is that anywhere? would that be of value? i don't actually know. i just find it hard to keep straight by looking at just the examples.